airbyte-cdk/python/unit_tests/sources/stream_reader/test_full_refresh_reader.py

```diff
@@ -0,0 +1,300 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from unittest.mock import Mock, call
+
+import pytest
+from airbyte_cdk.models import (
+    AirbyteCatalog,
+    AirbyteConnectionStatus,
+    AirbyteControlMessage,
+    AirbyteLogMessage,
+    AirbyteMessage,
+    AirbyteRecordMessage,
+    AirbyteStateMessage,
+    AirbyteTraceMessage,
+    ConnectorSpecification,
+    Level,
+    OrchestratorType,
+    Status,
+    SyncMode,
+    TraceType,
+)
+from airbyte_cdk.models import Type as MessageType
+from airbyte_cdk.sources.stream_reader.full_refresh_stream_reader import FullRefreshStreamReader
+from airbyte_cdk.sources.stream_reader.synchronous_full_refresh_reader import SyncrhonousFullRefreshReader
+from airbyte_cdk.sources.utils.schema_helpers import InternalConfig
+from airbyte_cdk.sources.utils.slice_logger import DebugSliceLogger
+
+_A_CURSOR_FIELD = ["NESTED", "CURSOR"]
+_DEFAULT_INTERNAL_CONFIG = InternalConfig()
+_STREAM_NAME = "STREAM"
+
+
+def _syncrhonous_reader():
+    return SyncrhonousFullRefreshReader(DebugSliceLogger())
+
+
+@pytest.mark.parametrize(
+    "reader",
+    [
+        pytest.param(_syncrhonous_reader(), id="synchronous_reader"),
```

Comment:
I'm parametrizing the tests so we can run the same ones on the concurrent stream reader to validate it respects the contract